internal/core/window.rs

```diff
@@ -596,8 +596,30 @@ impl WindowInner {
             self.had_popup_on_press.set(!self.active_popups.borrow().is_empty());
         }
 
-        let close_policy = self.top_close_policy();
-        let mut mouse_inside_popup = false;
+        let popup_to_close = self.active_popups.borrow().last().and_then(|popup| {
+            let mouse_inside_popup = || {
+                if let PopupWindowLocation::ChildWindow(coordinates) = &popup.location {
+                    event.position().map_or(true, |pos| {
+                        ItemTreeRc::borrow_pin(&popup.component)
+                            .as_ref()
+                            .item_geometry(0)
+                            .contains(pos - coordinates.to_vector())
+                    })
+                } else {
+                    false
+                }
+            };
+            match popup.close_policy {
+                PopupClosePolicy::CloseOnClick => {
+                    let mouse_inside_popup = mouse_inside_popup();
+                    (mouse_inside_popup && released_event && self.had_popup_on_press.get())
+                        || (!mouse_inside_popup && pressed_event)
+                }
+                PopupClosePolicy::CloseOnClickOutside => !mouse_inside_popup() && pressed_event,
+                PopupClosePolicy::NoAutoClose => false,
+            }
+            .then(|| popup.popup_id)
```

Comment:
I guess that could also be .then_some(popup.popup.id)

Reply:
Indeed

Reply:
```suggestion
            .then_some(popup.popup_id)
```